crates/pipeline_manager/src/db/mod.rs

```diff
@@ -2412,10 +2412,15 @@ impl Storage for ProjectDB {
                         &[&description, &config.to_yaml(), &service_id.0, &tenant_id.0],
                     )
                     .await
-                    .map_err(DBError::from)?;
+                    .map_err(DBError::from)?
             }
+        };
+        if modified_rows == 0 {
+            Err(DBError::UnknownService { service_id })
+        } else {
+            assert_eq!(modified_rows, 1);
```

Comment:
Log this as an error. Don't assert outside tests.

Reply:
Changed to:
```
if modified_rows != 1 {
    error!("More than one row ({modified_rows}) was modified when updating service {service_id}");
}
```

Reply:
Changed to:
```
if modified_rows > 0 {
    Ok(())
} else {
    Err(DBError::UnknownService { service_id })
}
```